onnxruntime/core/providers/xnnpack/xnnpack_execution_provider.cc

```diff
@@ -236,19 +237,36 @@
     }
     node_unit_supported_result[&node_unit] = request_node;
     if (request_node) {
+      auto required_register = [](const IndexedSubGraph& subgraph) -> bool {
+        if (subgraph.GetMetaDef() == nullptr) return false;
+
+        const std::string& op_type = subgraph.GetMetaDef()->name;
+        // layout insensitive ops registration, add more ops here if needed.
+        if (std::unordered_set<std::string>{"QLinearSoftmax"}.count(op_type)) {
+          return true;
+        }
+        return false;
+      };
+      auto register_layout_insensitive_ops = [&graph, &required_register](const IndexedSubGraph& subgraph) {
+        if(required_register(subgraph)){
+          auto temp_schema_ptr = function_utils::CreateSchema(graph.GetGraph(), subgraph);
+          // RegisterSchema may complain registration more than once, but it's fine.
```

Comment:
Would it be simpler to create these schemas upfront?

e.g. using CreateSchema as an example, do the same things around the time we call RegisteryKernels.

Something like
```c++
// e.g. add this to the xnnpack::Softmax class, similar to IsSoftmaxOnnxNodeSupported
static void AddSoftmaxInputOutputInfo(ONNX_NAMESPACE::OpSchema& schema) {
  // for simplicity create one list that covers all the valid types for the operator inputs and outputs.
  // we manually check if the node is supported in the EP, so if we ever create a node in our custom domain
  // we know it's supported. due to that the type constraints here don't really matter - they just need to be cover
  // all valid values. extra types don't hurt.
  schema.TypeConstraint("T", {"tensor(uint8)", "tensor(int8)"}, "...");
  schema.Input(0, "input", "", "T");
  schema.Output(0, "output", "", "T");
}

std::unique_ptr<ONNX_NAMESPACE::OpSchema>
CreateSchema(const std::string& name,
             const std::string& domain,
             int since_version,
             const std::function<void(ONNX_NAMESPACE::OpSchema&)>& add_input_output_info) {
  auto op_schema = std::make_unique<ONNX_NAMESPACE::OpSchema>();
  op_schema->SetName(name);
  op_schema->SetDomain(domain);
  op_schema->SetDoc("");
  op_schema->SinceVersion(since_version);

  add_input_output_info(*op_schema);

  op_schema->Finalize();

  return op_schema;
}

static void Usage() {
 // for each custom xnnpack kernel that we will register, create a schema
  auto schema = CreateSchema("QLinearSoftmax", "xnnpack", 1, AddSoftmaxInputOutputInfo);
}  

```